### PR TITLE
Add TLS support for Redis

### PR DESCRIPTION
**Pull request by dciabrin**
Ability to configure Redis with TLS endpoints and TLS replication for Redis HA.

Jira: [OSPRH-3567](https://issues.redhat.com//browse/OSPRH-3567)

### CODE REVIEW

File: pkg/redis/volumes.go [hunk 1]
```diff
@@ -102,10 +121,45 @@ func getVolumes(r *redisv1beta1.Redis) []corev1.Volume {
...
+		}
+	}
+
+	return vols
+}
+
+func getTLSVolumeMounts(r *redisv1beta1.Redis) []corev1.VolumeMount {
+	var vols []corev1.VolumeMount
+	if r.Spec.TLS.Enabled() {
+		svc := tls.Service{
```
Comment (olliewalsh) [inline pkg/redis/volumes.go:146]: why not use `r.Spec.TLS.GenericService.ToService()`?

File: pkg/redis/volumes.go [hunk 2]
```diff
@@ -102,10 +121,45 @@ func getVolumes(r *redisv1beta1.Redis) []corev1.Volume {
...
+		if r.Spec.TLS.Ca.CaBundleSecretName != "" {
+			caVolume := r.Spec.TLS.Ca.CreateVolume()
+			vols = append(vols, caVolume)
+		}
+	}
+
+	return vols
+}
+
+func getTLSVolumeMounts(r *redisv1beta1.Redis) []corev1.VolumeMount {
```
Comment (stuggi) [inline pkg/redis/volumes.go:143]: in the controller you changed it to `redisv1`, should we do this everywhere?

File: controllers/redis/redis_controller.go [hunk 3]
```diff
@@ -180,9 +201,37 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (result ct
 		return rbacResult, nil
 	}
 
+	// Hash of all resources that may cause a service restart
+	inputHashEnv := make(map[string]env.Setter)
+
+	// Check and hash inputs
+	var certHash, caHash string
+	specTLS := &instance.Spec.TLS
+	if err == nil && specTLS.Enabled() {
```
[the remark this answers is not in the document]
Comment (dciabrin) [reply controllers/redis/redis_controller.go:210]: You're correct, that's a leftover from a previous development. Thanks

File: pkg/redis/volumes.go (hunk 2 above)
Comment (dciabrin) [reply pkg/redis/volumes.go:143]: Right, I'll replicate the change everywhere, I think this is the pattern that other operator are using.

Comment (stuggi) [reply pkg/redis/volumes.go]: yes, then you do not have to update it everywhere when there is an api version bump

**Review by stuggi — APPROVED** [review]
/lgtm

**Pull request merged**